app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -15,15 +15,18 @@ module CollectionBehavior
       validates_with HasOneTitleValidator
       self.indexer = Hyrax::CollectionIndexer
 
+      class_attribute :index_collection_type_gid_as, writer: false
+      self.index_collection_type_gid_as = [:symbol]
+
       property :collection_type_gid, predicate: ::RDF::Vocab::SCHEMA.additionalType, multiple: false do |index|
-        index.as :symbol
+        index.as(*index_collection_type_gid_as)
       end
 
       after_find { |col| load_collection_type_instance(col) }
       # before_update { |col| validate_collection_type_gid(col) } # TODO: check that gid is not nil AND check that gid has not changed since it was read in (It is ok to go from nil to a value.)
     end
 
-    delegate :nestable?, :discoverable?, :sharable?, :allow_multiple_membership?, :require_membership?, :assigns_workflow?, :assigns_visibility?, to: :collection_type
+    delegate(*Hyrax::CollectionType.collection_type_predicate_methods, to: :collection_type)
```

Comment:
I went with the Ruby elements of style regarding predicate methods. (See 
http://pragmati.st/2012/03/24/the-elements-of-ruby-style-predicate-methods/) However, I will rename.